sync/downstream.py

```diff
@@ -612,28 +631,34 @@ def try_push_complete(git_gecko, git_wpt, try_push, sync):
             logger.info("Try push is not complete")
             return
 
-        try_push.validate_tasks()
-
-        logger.info("Try push %r for PR %s complete" % (try_push, sync.pr))
-        disabled = []
-        if not try_push.success():
-            if sync.affected_tests():
-                log_files = try_push.download_raw_logs()
-                if not log_files:
-                    raise ValueError("No log files found for try push %r" % try_push)
-                disabled = sync.update_metadata(log_files, stability=try_push.stability)
-            else:
-                env.bz.comment(sync.bug, ("The PR was not expected to affect any tests, "
-                                          "but the try push wasn't a success. Check the try "
-                                          "results for infrastructure issues"))
-                # TODO: consider marking the push an error here so that we can't land without manual
-                # intervention
-
-    if try_push.stability and disabled:
-        logger.info("The following tests were disabled:\n%s" % "\n".join(disabled))
-        # TODO notify relevant people about test expectation changes, stability
-        env.bz.comment(sync.bug, ("The following tests were disabled "
-                                  "based on stability try push:\n %s" % "\n".join(disabled)))
+        if not try_push.validate_tasks():
+            if len(sync.latest_busted_try_pushes()) > 5:
```

Comment:
If we call `next_action` here instead, we won't have to repeat the latest_busted_try_pushes logic.

Reply:
I'm not sure I see what you mean. `next_action` doesn't seem to contain this logic?

Reply:
Oh, sorry. So now I remember :) The problem is that next_action shouldn't be stateful, so it can't set the bug comment. But! I didn't actually finish fixing this right. I'll do that :)

Reply:
However, next_action does do everything except set the bug comment, so could call next_action + comment here, no?

@jgraham Anyway, the new changes so far lgtm, but you're not done yet? Let me know.

Reply:
The idea is that you can call next_action multiple times, but we presumably don't end up in `try_push_complete` unless we actually had a try push, so you don't end up commenting multiple times. It's probably not perfect, but I think it's not totally wrong?

Anyway I think this PR is "complete" but there are others based on it.